include/clang/AST/Decl.h

```diff
@@ -731,6 +734,25 @@ class DeclaratorDecl : public ValueDecl {
 
   friend class ASTDeclReader;
   friend class ASTDeclWriter;
+
+  // Checked C bounds information
+
+  /// \brief Return true if this declaration has bounds declared for it.
+  bool hasBoundsExpr() const;
+
+  /// \brief The declared bounds for this declaration. For function
+  /// declarations, this is the return bounds of the function. Null if no
+  /// bounds have been declared.
+  const BoundsExpr *getBoundsExpr() const {
```

Comment:
I don't there is a standard, but I've seen asserts using dynamic types that guard against the misuse of members for derived classes.

---
In reply to: [72876100](https://github.com/Microsoft/checkedc-clang/pull/23#discussion_r72876100) [](ancestors = 72876100)